gpflow/models/svgp.py

```diff
@@ -170,3 +175,75 @@ def predict_f(self, Xnew: InputData, full_cov=False, full_output_cov=False) -> M
         )
         # tf.debugging.assert_positive(var)  # We really should make the tests pass with this here
         return mu + self.mean_function(Xnew), var
+
+
+class NewSVGP(OldSVGP):
+    """
+    Adds posterior() method and uses different math ordering for predict_f
+    """
+
+    def __init__(
+        self,
+        kernel,
+        likelihood,
+        inducing_variable,
+        *,
+        mean_function=None,
+        num_latent_gps: int = 1,
+        q_diag: bool = False,
+        q_mu=None,
+        q_sqrt=None,
+        whiten: bool = True,
+        num_data=None,
+    ):
+        """
+        - kernel, likelihood, inducing_variables, mean_function are appropriate
+          GPflow objects
+        - num_latent_gps is the number of latent processes to use, defaults to 1
+        - q_diag is a boolean. If True, the covariance is approximated by a
+          diagonal matrix.
+        - whiten is a boolean. If True, we use the whitened representation of
+          the inducing points.
+        - num_data is the total number of observations, defaults to X.shape[0]
+          (relevant when feeding in external minibatches)
+        """
+        # init the super class, accept args
+        super().__init__(
+            kernel,
+            likelihood,
+            inducing_variable,
+            mean_function=mean_function,
+            num_latent_gps=num_latent_gps,
+            q_diag=q_diag,
+            q_mu=q_mu,
+            q_sqrt=q_sqrt,
+            whiten=whiten,
+            num_data=num_data,
+        )
+
+    def posterior(self):
+        """
+        Create the Posterior object which contains precomputed matrices for
+        faster prediction when wrapped inside tf.function()
+        """
+        return Posterior(
+            self.kernel,
+            self.inducing_variable,
+            self.q_mu,
+            self.q_sqrt,
+            whiten=self.whiten,
+            mean_function=self.mean_function,
+        )
+
+    def predict_f(self, Xnew: InputData, full_cov=False, full_output_cov=False) -> MeanAndVariance:
+        """
+        For backwards compatibility. We have to update posterior's cache here so we have acces to
+        the latest values during training.
+        For faster (cached) prediction, predict directly from the posterior object, i.e.,:
+            model.posterior.predict_f(Xnew, ...)
+        """
+        return self.posterior().predict_f(Xnew, full_cov=full_cov, full_output_cov=full_output_cov)
+
+
+class SVGP(NewSVGP):
```

Comment:
Why not keep SVGP as is, and let NewSVGP inherit from SVGP immediately?

Reply:
What @stefanosele said - this runs all pre-existing tests (and importantly the SVGP vs VGP method equivalence comparison) against our new code. Subclassing instead of aliasing `SVGP = NewSVGP` (which I actually started out with) means that `SVGP(...).__class__.__name__` is "SVGP" and we don't have to touch all the printing tests.